### DISCUSSION TITLE
<Denoise start> as an xyz grid parameter

### DISCUSSION THREAD

**Discussion by SAC020** [vladmandic/sdnext#3594] [General]
Hi,

**Denoise start** is not a predefined parameter in xyz grid. However I remember a recent changelog mentioning that some parameters can be used in xyz following a certain syntax.

Is it possible to xyz by **denoise start**? If yes, what is the parameter name / syntax to do so?

Thanks!

**Comment by vladmandic** [comment]
![image](https://github.com/user-attachments/assets/3d138682-28e4-4a87-ab03-255b4a0a4af8)

btw, its not some, its all parameters, you just have to know their internal name.
you can take a look at `modules/processing_class.py:StableDiffusionProcessing` for list.

![image](https://github.com/user-attachments/assets/60c12afe-0a89-4a95-a723-d2c48c3a6b2d)

**Reply by SAC020** [reply]
Thank you. I sort of anticipated that but I didn't know where to look for the parameter names. Excellent stuff!

**Comment by SAC020** [comment]
I am not sure we are talking about the same parameter

I am looking for denoise start, in img2img

![image](https://github.com/user-attachments/assets/3c5e1e80-3c03-49e9-8d6f-d2de87b61824)

I've tried refiner start, returns identical results regardless of setting. I am not using, nor enabled any refiner.

I've searched the file, I can't find anything related to denoise start.

**Reply by vladmandic** [reply]
we're talking about the same thing

**Reply by vladmandic** [reply]
![image](https://github.com/user-attachments/assets/6549081d-f343-4fb2-bcdd-00b2cabd6781)

**Comment by SAC020** [comment]
Thanks. It works. What misled me is that on Flux it produces identical outputs, but on SDXL it does have an effect

![tmpar5ydh70](https://github.com/user-attachments/assets/fcc0275d-abfc-4ab9-937f-ab6765b11966)

![tmpi1i04yr9](https://github.com/user-attachments/assets/4524b428-85c3-4418-addb-3e64b85f5bac)

**Comment by SAC020** [comment]
Hi,

Can I run an xyz grid in control, with processor on x and controlnet on y? I'm trying to find best combos and there are a lot of processors and controlnets available for sdxl (which is a good thing which I want to leverage)

I looked in `modules/processing_class.py:StableDiffusionProcessing` but could not find any parameter that could be related, and there is nothing about controlnet in the xyz dropdown lists either (unless I'm misreading it)

Thanks

**Reply by SAC020** [reply]
What is the parameter name for "attention guidance" that I can use in xyz?

<img width="316" height="98" alt="image" src="https://github.com/user-attachments/assets/721e393a-da70-488e-911c-ef4139f63bb1" />

I see it also translates into "CFG true". I cannot find either "attention guidance" or "cfg true" (or similar) in `modules/processing_class.py:StableDiffusionProcessing`

Parameters: Steps: 40| Size: 1024x1024| Seed: 2297150924| CFG scale: 4| **CFG true: 6**| App: SD.Next| Version: c2bc47e| Pipeline: Flux2Pipeline| Operations: txt2img|

Thanks

**Comment by vladmandic** [comment]
attention guidance is internally called `cfg_scale`
